File: Code/Source/Estimatorx.Web/app/models/QueryRequest.ts
/// <reference path="../_ref.ts" />

module Estimatorx {
  "use strict";

  export interface IQueryRequest {
    Page?: number;
    PageSize?: number;

    Sort?: string;
    Descending?: boolean;

    Filter?: any;

  }

  export interface IQuerySearch extends IQueryRequest {
    Search?: string;
    Organization?: string;
  }

}

File: src/EstimatorX.Shared/Models/QueryRequest.cs
namespace EstimatorX.Shared.Models;

public class QueryRequest
{
    public int? Page { get; set; } = 1;

    public int? PageSize { get; set; } = 20;

    public string Sort { get; set; }

    public bool? Descending { get; set; }

    public string Search { get; set; }

    public string Organization { get; set; }
}
